client/main.go

```diff
@@ -468,25 +467,35 @@ func main() {
 	myApp.Run(os.Args)
 }
 
-func scavenger(ch chan *smux.Session, config *Config) {
+func scavenger(ch chan timedSession, config *Config) {
+	// When AutoExpire is set to 0 (default), sessionList will keep empty.
+	// Then this routine won't need to do anything; thus just terminate it.
+	if config.AutoExpire <= 0 {
+		return
+	}
+
 	ticker := time.NewTicker(time.Second)
 	defer ticker.Stop()
 	var sessionList []timedSession
 	for {
 		select {
-		case sess := <-ch:
+		case item := <-ch:
 			sessionList = append(sessionList, timedSession{
-				sess,
-				time.Now().Add(time.Duration(config.ScavengeTTL+config.AutoExpire) * time.Second)})
```

Comment:
Here in scavenger(), we better not depends on `config. AutoExpire`, instead to fetch the expire info from them session item. So that a change to the `timedSession. expiryDate` built logic won't need to touch the code in scavenger too.